### PR TITLE
Move Druid tests to BaseConnectorTest

### PR DESCRIPTION
**Pull request by wendigo**

### CODE REVIEW

File: plugin/trino-druid/src/test/java/io/trino/plugin/druid/BaseDruidConnectorTest.java [hunk 1]
```diff
@@ -115,6 +115,48 @@ public void destroy()
         }
     }
 
+    @Override
```
Comment (hashhar) [inline plugin/trino-druid/src/test/java/io/trino/plugin/druid/BaseDruidConnectorTest.java:118]: Let's remove these and use `hasBehavior`. See https://github.com/trinodb/trino/pull/7274 for the idea.

**Review by hashhar — COMMENTED** [review]
Changes look good.

The commit message should also mention the removal of DruidErrorCode (should ideally be a separate commit).

File: plugin/trino-druid/src/test/java/io/trino/plugin/druid/BaseDruidConnectorTest.java (hunk 1 above)
Comment (wendigo) [reply plugin/trino-druid/src/test/java/io/trino/plugin/druid/BaseDruidConnectorTest.java:118]: Yeah, but that PR is required for mine to work.

Comment (wendigo) [reply plugin/trino-druid/src/test/java/io/trino/plugin/druid/BaseDruidConnectorTest.java:119]: I've rebased on top of that and pushed `hasBehaviour` as a fixup

File: plugin/trino-druid/src/main/java/io/trino/plugin/druid/DruidJdbcClient.java [hunk 2]
```diff
@@ -222,66 +223,66 @@ public boolean isLimitGuaranteed(ConnectorSession session)
...
     {
-        throw new TrinoException(DruidErrorCode.DRUID_DDL_NOT_SUPPORTED, "DDL operations are not supported in the Trino Druid connector");
+        throw new TrinoException(NOT_SUPPORTED, "This connector does not support creating schemas");
     }
 
     @Override
     public void dropSchema(ConnectorSession session, String schemaName)
     {
-        throw new TrinoException(DruidErrorCode.DRUID_DDL_NOT_SUPPORTED, "DDL operations are not supported in the Trino Druid connector");
+        throw new TrinoException(NOT_SUPPORTED, "DThis connector does not support dropping schemas");
```
Comment (losipiuk) [inline plugin/trino-druid/src/main/java/io/trino/plugin/druid/DruidJdbcClient.java:286]: s/DThis/This

**Review by losipiuk — APPROVED** [review]

File: plugin/trino-druid/src/main/java/io/trino/plugin/druid/DruidJdbcClient.java (hunk 2 above)
Comment (wendigo) [reply plugin/trino-druid/src/main/java/io/trino/plugin/druid/DruidJdbcClient.java]: Good catch :)

**Review by hashhar — APPROVED** [review]

**Pull request merged**